eudemo/eudemo/maintenance/lower_port/duct_designer.py

```diff
@@ -215,8 +223,12 @@ def _calc_y_point(x_point):
 
             return x_len * np.tan(self._half_beta)
 
-        ib_inner_y = _calc_y_point(ib_div_pt_padded[0]) - self.wall_tk
-        ob_inner_y = _calc_y_point(ob_div_pt_padded[0]) - self.wall_tk
+        # TODO alternative limit
+        flat_y_max = self.port_width / 2
+        ib_inner_y = min(flat_y_max, _calc_y_point(ib_div_pt_padded[0]) - self.wall_tk)
+        ob_inner_y = min(flat_y_max, _calc_y_point(ob_div_pt_padded[0]) - self.wall_tk)
+        # ib_inner_y = _calc_y_point(ib_div_pt_padded[0]) - self.wall_tk
```

Comment:
Yeah okay this makes sense. It won't fill up all the space, only up to what it needs to (i.e. port_width/2)